Version 1.0.4

johnmartins opened the pull request:
- [x] File Access Control
- [x] File removal
- [x] Improved MySQL Statement Builder robustness
- [x] Remove mysqlutils from libs and import from pypi instead
- [x] Fix failing CVS tests
- [x] Search for users
- [x] Sort user lists
- [x] Update projects

Closes #83 
Closes #92

EppChops commented on sedbackend/apps/core/files/dependencies.py:
```diff
@@ -0,0 +1,28 @@
+
+        # Run subproject access check
+        subproject = impl_get_subproject_by_id(subproject_id)
+        return SubProjectAccessChecker.check_user_subproject_access(subproject, AccessLevel.list_can_read(), user_id)
```
Hi!

Is is just me or will it automatically be always list_can_read() when accessing a file here?
I.e if I was to use this for deletion (where we probably should use list_can_edit()), will it be restricted to only those who are allowed to edit?

johnmartins replied:
When downloading or accessing the file, then it will always be list_can_read. In your case, where the user wants to delete the file, it should probably be list_can_edit, as you say.

johnmartins replied:
Oh, now I see what you mean. I'll fix it.

johnmartins replied:
@EppChops the issue has now been resolved. Pull the latest version of this branch.

EppChops replied:
Perfect thank you!

johnmartins commented:
@EppChops @obennet I've removed mysqlutils from the project and turned it into an independent pypi package. I've updated all instances where it was used (also in cvs app). Also, please have a look at the failing CVS tests for this version release.

johnmartins commented:
@obennet @massimop-tech 
The features for 1.0.4 are now completed so I will be merging this into main some time next week. This will result in a short downtime for the server. It will not result in any data loss. I'll send an e-mail before performing the update.

@obennet We will put your fixes to the simulation database communication in a dedicated patch.

johnmartins commented:
@obennet the multiprocessing tests are failing again. Can you take a look at the failing tests? I've added some functionality to the projects API, but it shouldn't be breaking (all project related tests work fine)

obennet commented:
I reran the tests and it worked. The random tests that we have for the simulations are not very good because it sometimes produces incorrect sets of data which makes the test fail. The code runs as it should I just have to rewrite the tests.
